src/hooks/useInteractive.ts

```diff
@@ -1,18 +1,21 @@
-import { useRef, MouseEvent, useState } from 'react'
+import { useRef, MouseEvent } from 'react'
 import { useSelector } from 'react-redux'
 import useDispatch from './useDispatch'
 import { useDrag } from 'react-dnd'
-import { getSelectedComponentId } from '../core/selectors/components'
+import {
+  getSelectedComponentId,
+  getHoveredId,
+} from '../core/selectors/components'
 import { getShowLayout } from '../core/selectors/app'
 
 export const useInteractive = (
   component: IComponent,
   enableVisualHelper: boolean = false,
 ) => {
   const dispatch = useDispatch()
-  const [hover, setHover] = useState(false)
   const showLayout = useSelector(getShowLayout)
   const selectedId = useSelector(getSelectedComponentId)
+  const hoveredId = useSelector(getHoveredId)
```

Comment:
We should use a selector that checks the componentid to avoid useless re-renders